database/dbService.js

```diff
@@ -0,0 +1,84 @@
+var mysql = require('mysql');
+var keys = require('../config/keys');
+var queries = require('./queries');
+
+var connection;
+module.exports = {
+    /* Inserts a given user model given by Google OAuth */
+    insertUser: function(user, callback) {
+        this.tryConnect().getConnection(function(err, con) {
+            var sql = queries.insertUser;
+            con.query(sql, [user.id, user.gender, user.name.givenName, user.name.familyName, user.emails[0].value], function (err, result) {
+                con.release();
+                if (err) throw err;
+                callback(result.insertId);
+            });
+        });
+    },
+
+    /* Retrieves a User model by GoogleID */
+    getUserByGoogleID: function(googleID, callback) {
+        this.tryConnect().getConnection(function(err, con) {
+            var sql = queries.getUserByGoogleID;
+            con.query(sql, googleID, function (err, result) {
+                con.release();
+                if (err) throw err;
+                // Call the callback function in the caller of this method so we can do something with this "result"
+                return callback(result); // [] if not found
+            });
+        });
+    },
+
+    /* Retrieves a User model by ID */
+    getUserByID: function(ID, callback) {
+        this.tryConnect().getConnection(function(err, con) {
+            var sql = queries.getUserByID;
+            con.query(sql, ID, function (err, result) {
+                con.release();
+                if (err) throw err;
+                // Call the callback function in the caller of this method so we can do something with this "result"
+                return callback(result); // [] if not found
+            });
+        });
+    },
+
+    tryConnect: function() {
+        // Singleton: if connection is already established, return it
```

Comment:
This is the method to connect to the MySQL database and also has handling in case the server closes the connection, which happens sometimes